### PR TITLE
Statfs under Solaris/Illumos needs 4 arguments

### PR DESCRIPTION
**Pull request by urknall**

### CODE REVIEW

File: CPAN/EV_fstyp.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+-      else if (!statfs (w->path, &sfs)
+-               && (sfs.f_type == 0x1373 /* devfs */
+-                   || sfs.f_type == 0xEF53 /* ext2/3 */
+-                   || sfs.f_type == 0x3153464a /* jfs */
+-                   || sfs.f_type == 0x52654973 /* reiser3 */
+-                   || sfs.f_type == 0x01021994 /* tempfs */
+-                   || sfs.f_type == 0x58465342 /* xfs */))
++      else if (!statfs (w->path, &sfs, len, fstyp)
++               && (sfs.f_fstyp == 0x1373 /* devfs */
++                   || sfs.f_fstyp == 0xEF53 /* ext2/3 */
```
Comment (mherger) [inline CPAN/EV_fstyp.patch:25]: Do you have any information about why this change is required? This looks like a change to an underlying module. How come we have to change this patch, when we do not the module's source?

File: CPAN/buildme.sh [hunk 2]
```diff
@@ -799,6 +799,11 @@ function build {
                     patch -p0 < ../EV-fixes.patch # patch to disable pthreads and one call to SvREADONLY
                 fi
             fi
+	    if [ "$OS" = "SunOS" ]; then
+	    	cd libev
+		patch -p0 < ../../EV_fstyp.patch
+		cd ..
+	    fi
```
Comment (mherger) [inline CPAN/buildme.sh]: Please make sure you use the same style indentation as the rest (spaces vs. tabs).

**Review by mherger — CHANGES_REQUESTED** [review]
Could you please
- squash changes
- describe the changes and the motivation for them
- whitespace cleanup
- give some details about the change to the EV patch

The last one isn't specific to your target OS, but is applied to all platforms, isn't it? Therefore it's particularly important to be careful what we do. I don't understand how we'd have to change calling an interface, when the interface itself doesn't seem to change. Did you miss to commit a file?

File: CPAN/EV_fstyp.patch (hunk 1 above)
Comment (urknall) [reply CPAN/EV_fstyp.patch:25]: The patch was not changed, i added a new one for solaris/illumos and its only applied for that os. Other operating systems were not affected. 

Statfs on Solaris exists for backwards compatibility, but takes four parameters. The prototype is

/usr/include/sys/statfs.h:int statfs(const char *, struct statfs *, int, int)

File: CPAN/buildme.sh (hunk 2 above)
Comment (urknall) [reply CPAN/buildme.sh]: whitespace cleanup done

File: CPAN/EV_fstyp.patch (hunk 1 above)
Comment (mherger) [reply CPAN/EV_fstyp.patch:25]: Ok, got it. Could you add some hint about the OS in the patch's name? Thanks!

Comment (urknall) [reply CPAN/EV_fstyp.patch:25]: changed...
